lib/puppet/util/selinux.rb

```diff
@@ -65,13 +65,18 @@ def parse_selinux_context(component, context)
     unless context =~ /^([^\s:]+):([^\s:]+):([^\s:]+)(?::([\sa-zA-Z0-9:,._-]+))?$/
       raise Puppet::Error, "Invalid context to parse: #{context}"
     end
-    ret = {
-      :seluser => $1,
-      :selrole => $2,
-      :seltype => $3,
-      :selrange => $4,
-    }
-    ret[component]
+    case component
+    when :seluser
+      $1
```

Comment:
wow this syntax really surprised me - I don't think I've ever seen using these global $<int> variables for regex matches before in ruby.